player.py

```diff
@@ -125,29 +133,51 @@ def __iter__(self):
 
 
 class CaptainState(State):
-    def __init__(self, player, game):
-        super().__init__(player, game)
-        self.board_str = player.get_board_str(game)
+    def __init__(self, can_act, is_game_stopped, board_str):
+        super().__init__(can_act, is_game_stopped)
+        self.board_str = board_str
+
+    @classmethod
+    def from_player(cls, player, game):
+        state = State.from_player(player, game)
```

Comment:
Instead of calling `State.from_player` you should call `super().from_player`

Reply:
it doesnt work with `super().from_player`, the `cls =State` parameter isn't sent to the function